### PR TITLE
Support smallest_time_grain and inf time range

### LINKED ISSUE #1766: implement minimum timegrain in metrics configuration
We've taken a moment to reconsider the work needed for configurable time defaults for `0.22`, and decided to change our approach. We'll keep the default timerange, but
1. we'll remove the default time grain & available timegrain options
2. we'll add a minimum time grain option

cc @djbarnwal for awareness

### PR DESCRIPTION
**Pull request by begelundmuller**
Closes #1766

### CODE REVIEW

**Comment by begelundmuller** [comment]
Closes https://github.com/rilldata/rill-developer/issues/1766

**Comment by hamilton** [comment]
Tagging @djbarnwal for review so he can assess if this works as expected. Thanks for the PR @begelundmuller

File: web-common/src/features/metrics-views/metrics-internal-store.ts [hunk 1]
```diff
@@ -13,8 +13,7 @@ export interface MetricsConfig {
   display_name: string;
   description: string;
   timeseries: string;
-  time_grains?: Array<string>;
-  default_time_grain?: Array<string>;
+  smallest_time_grain?: Array<string>;
```
Comment (djbarnwal) [inline web-common/src/features/metrics-views/metrics-internal-store.ts:16]: This should just be string right?

**Review by djbarnwal — APPROVED** [review]
Functionality wise, looks good to me. Added a small nit regarding the type for `smallest_time_grain`.

File: web-common/src/features/metrics-views/metrics-internal-store.ts (hunk 1 above)
Comment (begelundmuller) [reply web-common/src/features/metrics-views/metrics-internal-store.ts]: Yeah, good catch! Fixed it.

**Pull request merged**